include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -500,9 +506,9 @@ namespace VectorTools
     // internal implementation
     const auto function_map = [&function](
                                 const typename DoFHandlerType<dim, spacedim>::active_cell_iterator &)
-                              -> const Function<spacedim, typename VectorType::value_type> &
+                              -> std::pair<bool, const Function<spacedim, typename VectorType::value_type> *>
     {
-      return function;
+      return std::make_pair(true, &function);
     };
```

Comment:
I know that this signature is a bit ugly
  `std::pair<bool, const Function<spacedim, typename VectorType::value_type> *>`

An alternative would be to provide two lambda captures - one for a function value and one for the boolean. But that doesn't really make things prettier.

The boolean is necessary because `interpolate_based_on_material_id` might simply skip cells (if there is no map for the current material id).

Reply:
Oh, on second thought. I could encode the bool in the pointer. I.e. shortcut if nullptr is encountered.
(Sometimes I am blind.)